var formLib=function(o){var n={};function a(e){if(n[e])return n[e].exports;var t=n[e]={i:e,l:!1,exports:{}};return o[e].call(t.exports,t,t.exports,a),t.l=!0,t.exports}return a.m=o,a.c=n,a.d=function(e,t,o){a.o(e,t)||Object.defineProperty(e,t,{enumerable:!0,get:o})},a.r=function(e){"undefined"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:"Module"}),Object.defineProperty(e,"__esModule",{value:!0})},a.t=function(t,e){if(1&e&&(t=a(t)),8&e)return t;if(4&e&&"object"==typeof t&&t&&t.__esModule)return t;var o=Object.create(null);if(a.r(o),Object.defineProperty(o,"default",{enumerable:!0,value:t}),2&e&&"string"!=typeof t)for(var n in t)a.d(o,n,function(e){return t[e]}.bind(null,n));return o},a.n=function(e){var t=e&&e.__esModule?function(){return e.default}:function(){return e};return a.d(t,"a",t),t},a.o=function(e,t){return Object.prototype.hasOwnProperty.call(e,t)},a.p="",a(a.s=48)}({1:function(e,t,o){"use strict";function n(e,t,o,n,a,i,s,r){var d,l="function"==typeof e?e.options:e;if(t&&(l.render=t,l.staticRenderFns=o,l._compiled=!0),n&&(l.functional=!0),i&&(l._scopeId="data-v-"+i),s?(d=function(e){(e=e||this.$vnode&&this.$vnode.ssrContext||this.parent&&this.parent.$vnode&&this.parent.$vnode.ssrContext)||"undefined"==typeof __VUE_SSR_CONTEXT__||(e=__VUE_SSR_CONTEXT__),a&&a.call(this,e),e&&e._registeredComponents&&e._registeredComponents.add(s)},l._ssrRegister=d):a&&(d=r?function(){a.call(this,this.$root.$options.shadowRoot)}:a),d)if(l.functional){l._injectStyles=d;var c=l.render;l.render=function(e,t){return d.call(t),c(e,t)}}else{var h=l.beforeCreate;l.beforeCreate=h?[].concat(h,d):[d]}return{exports:e,options:l}}o.d(t,"a",function(){return n})},23:function(e,t,o){},41:function(e){e.exports={athaliana:{geneontology:[{idtype:"entrezgene",name:"Biological_Process",description:"The gene ontology biological process database was downloaded from http://www.geneontology.org/."},{idtype:"entrezgene",name:"Biological_Process_noRedundant",description:"The gene ontology biological process database was downloaded from http://www.geneontology.org/. Then, we only contain the non-redundant categories by selecting the most general categories in each branch of the GO DAG structure from all categories with the number of annotated genes from 20 to 500."},{idtype:"entrezgene",name:"Cellular_Component",description:"The gene ontology cellular component database was downloaded from http://www.geneontology.org/."},{idtype:"entrezgene",name:"Cellular_Component_noRedundant",description:"The gene ontology cellular component database was downloaded from http://www.geneontology.org/. Then, we only contain the non-redundant categories by selecting the most general categories in each branch of the GO DAG structure from all categories with the number of annotated genes from 20 to 500."},{idtype:"entrezgene",name:"Molecular_Function",description:"The gene ontology molecular function database was downloaded from http://www.geneontology.org/."},{idtype:"entrezgene",name:"Molecular_Function_noRedundant",description:"The gene ontology molecular function database was downloaded from http://www.geneontology.org/. Then, we only contain the non-redundant categories by selecting the most general categories in each branch of the GO DAG structure from all categories with the number of annotated genes from 20 to 500."}],pathway:[{idtype:"entrezgene",name:"KEGG",description:"The KEGG pathway database was downloaded from http://www.kegg.jp/."},{idtype:"entrezgene",name:"Wikipathway",description:"The Wikipathway database was downloaded from http://www.wikipathway.org/."}],network:[{idtype:"entrezgene",name:"PPI_BIOGRID",description:"The protein-protein interaction (PPI) network was downloaded from BIOGRID (https://thebiogrid.org/). Then, we used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules."}],disease:[],drug:[],phenotype:[],chromosomalLocation:[{idtype:"entrezgene",name:"CytogeneticBand",description:""}],"community-contributed":[],others:[]},btaurus:{geneontology:[{idtype:"entrezgene",name:"Biological_Process",description:"The gene ontology biological process database was downloaded from http://www.geneontology.org/."},{idtype:"entrezgene",name:"Biological_Process_noRedundant",description:"The gene ontology biological process database was downloaded from http://www.geneontology.org/. Then, we only contain the non-redundant categories by selecting the most general categories in each branch of the GO DAG structure from all categories with the number of annotated genes from 20 to 500."},{idtype:"entrezgene",name:"Cellular_Component",description:"The gene ontology cellular component database was downloaded from http://www.geneontology.org/."},{idtype:"entrezgene",name:"Cellular_Component_noRedundant",description:"The gene ontology cellular component database was downloaded from http://www.geneontology.org/. Then, we only contain the non-redundant categories by selecting the most general categories in each branch of the GO DAG structure from all categories with the number of annotated genes from 20 to 500."},{idtype:"entrezgene",name:"Molecular_Function",description:"The gene ontology molecular function database was downloaded from http://www.geneontology.org/."},{idtype:"entrezgene",name:"Molecular_Function_noRedundant",description:"The gene ontology molecular function database was downloaded from http://www.geneontology.org/. Then, we only contain the non-redundant categories by selecting the most general categories in each branch of the GO DAG structure from all categories with the number of annotated genes from 20 to 500."}],pathway:[{idtype:"entrezgene",name:"KEGG",description:"The KEGG pathway database was downloaded from http://www.kegg.jp/."},{idtype:"entrezgene",name:"Panther",description:"The PANTHER pathway database was downloaded from http://www.pantherdb.org/pathway/."},{idtype:"entrezgene",name:"Reactome",description:"The Reactome pathway database was downloaded from http://www.reactome.org/."},{idtype:"entrezgene",name:"Wikipathway",description:"The Wikipathway database was downloaded from http://www.wikipathway.org/."}],network:[{idtype:"entrezgene",name:"CORUM",description:"The core complex subunits were downloaded from CORUM database (https://mips.helmholtz-muenchen.de/corum/#)."},{idtype:"entrezgene",name:"CORUMA",description:""},{idtype:"entrezgene",name:"PPI_BIOGRID",description:"The protein-protein interaction (PPI) network was downloaded from BIOGRID (https://thebiogrid.org/). Then, we used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules."},{idtype:"entrezgene",name:"Transcription_Factor_target",description:"The relationship between transcription factor and its targets was downloaded from MsigDB (http://software.broadinstitute.org/gsea/msigdb)."},{idtype:"entrezgene",name:"miRNA_target",description:"The relationship between miRNA and its targets was downloaded from MsigDB (http://software.broadinstitute.org/gsea/msigdb)."}],disease:[],drug:[],phenotype:[],chromosomalLocation:[{idtype:"entrezgene",name:"CytogeneticBand",description:""}],"community-contributed":[],others:[]},celegans:{geneontology:[{idtype:"entrezgene",name:"Biological_Process",description:"The gene ontology biological process database was downloaded from http://www.geneontology.org/."},{idtype:"entrezgene",name:"Biological_Process_noRedundant",description:"The gene ontology biological process database was downloaded from http://www.geneontology.org/. Then, we only contain the non-redundant categories by selecting the most general categories in each branch of the GO DAG structure from all categories with the number of annotated genes from 20 to 500."},{idtype:"entrezgene",name:"Cellular_Component",description:"The gene ontology cellular component database was downloaded from http://www.geneontology.org/."},{idtype:"entrezgene",name:"Cellular_Component_noRedundant",description:"The gene ontology cellular component database was downloaded from http://www.geneontology.org/. Then, we only contain the non-redundant categories by selecting the most general categories in each branch of the GO DAG structure from all categories with the number of annotated genes from 20 to 500."},{idtype:"entrezgene",name:"Molecular_Function",description:"The gene ontology molecular function database was downloaded from http://www.geneontology.org/."},{idtype:"entrezgene",name:"Molecular_Function_noRedundant",description:"The gene ontology molecular function database was downloaded from http://www.geneontology.org/. Then, we only contain the non-redundant categories by selecting the most general categories in each branch of the GO DAG structure from all categories with the number of annotated genes from 20 to 500."}],pathway:[{idtype:"entrezgene",name:"KEGG",description:"The KEGG pathway database was downloaded from http://www.kegg.jp/."},{idtype:"entrezgene",name:"Panther",description:"The PANTHER pathway database was downloaded from http://www.pantherdb.org/pathway/."},{idtype:"entrezgene",name:"Reactome",description:"The Reactome pathway database was downloaded from http://www.reactome.org/."},{idtype:"entrezgene",name:"Wikipathway",description:"The Wikipathway database was downloaded from http://www.wikipathway.org/."}],network:[{idtype:"entrezgene",name:"PPI_BIOGRID",description:"The protein-protein interaction (PPI) network was downloaded from BIOGRID (https://thebiogrid.org/). Then, we used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules."},{idtype:"entrezgene",name:"Transcription_Factor_target",description:"The relationship between transcription factor and its targets was downloaded from MsigDB (http://software.broadinstitute.org/gsea/msigdb)."},{idtype:"entrezgene",name:"miRNA_target",description:"The relationship between miRNA and its targets was downloaded from MsigDB (http://software.broadinstitute.org/gsea/msigdb)."}],disease:[],drug:[],phenotype:[],chromosomalLocation:[{idtype:"entrezgene",name:"CytogeneticBand",description:""}],"community-contributed":[],others:[]},cfamiliaris:{geneontology:[{idtype:"entrezgene",name:"Biological_Process",description:"The gene ontology biological process database was downloaded from http://www.geneontology.org/."},{idtype:"entrezgene",name:"Biological_Process_noRedundant",description:"The gene ontology biological process database was downloaded from http://www.geneontology.org/. Then, we only contain the non-redundant categories by selecting the most general categories in each branch of the GO DAG structure from all categories with the number of annotated genes from 20 to 500."},{idtype:"entrezgene",name:"Cellular_Component",description:"The gene ontology cellular component database was downloaded from http://www.geneontology.org/."},{idtype:"entrezgene",name:"Cellular_Component_noRedundant",description:"The gene ontology cellular component database was downloaded from http://www.geneontology.org/. Then, we only contain the non-redundant categories by selecting the most general categories in each branch of the GO DAG structure from all categories with the number of annotated genes from 20 to 500."},{idtype:"entrezgene",name:"Molecular_Function",description:"The gene ontology molecular function database was downloaded from http://www.geneontology.org/."},{idtype:"entrezgene",name:"Molecular_Function_noRedundant",description:"The gene ontology molecular function database was downloaded from http://www.geneontology.org/. Then, we only contain the non-redundant categories by selecting the most general categories in each branch of the GO DAG structure from all categories with the number of annotated genes from 20 to 500."}],pathway:[{idtype:"entrezgene",name:"KEGG",description:"The KEGG pathway database was downloaded from http://www.kegg.jp/."},{idtype:"entrezgene",name:"Panther",description:"The PANTHER pathway database was downloaded from http://www.pantherdb.org/pathway/."},{idtype:"entrezgene",name:"Reactome",description:"The Reactome pathway database was downloaded from http://www.reactome.org/."},{idtype:"entrezgene",name:"Wikipathway",description:"The Wikipathway database was downloaded from http://www.wikipathway.org/."}],network:[{idtype:"entrezgene",name:"PPI_BIOGRID",description:"The protein-protein interaction (PPI) network was downloaded from BIOGRID (https://thebiogrid.org/). Then, we used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules."},{idtype:"entrezgene",name:"Transcription_Factor_target",description:"The relationship between transcription factor and its targets was downloaded from MsigDB (http://software.broadinstitute.org/gsea/msigdb)."},{idtype:"entrezgene",name:"miRNA_target",description:"The relationship between miRNA and its targets was downloaded from MsigDB (http://software.broadinstitute.org/gsea/msigdb)."}],disease:[],drug:[],phenotype:[],chromosomalLocation:[{idtype:"entrezgene",name:"CytogeneticBand",description:""}],"community-contributed":[],others:[]},drerio:{geneontology:[{idtype:"entrezgene",name:"Biological_Process",description:"The gene ontology biological process database was downloaded from http://www.geneontology.org/."},{idtype:"entrezgene",name:"Biological_Process_noRedundant",description:"The gene ontology biological process database was downloaded from http://www.geneontology.org/. Then, we only contain the non-redundant categories by selecting the most general categories in each branch of the GO DAG structure from all categories with the number of annotated genes from 20 to 500."},{idtype:"entrezgene",name:"Cellular_Component",description:"The gene ontology cellular component database was downloaded from http://www.geneontology.org/."},{idtype:"entrezgene",name:"Cellular_Component_noRedundant",description:"The gene ontology cellular component database was downloaded from http://www.geneontology.org/. Then, we only contain the non-redundant categories by selecting the most general categories in each branch of the GO DAG structure from all categories with the number of annotated genes from 20 to 500."},{idtype:"entrezgene",name:"Molecular_Function",description:"The gene ontology molecular function database was downloaded from http://www.geneontology.org/."},{idtype:"entrezgene",name:"Molecular_Function_noRedundant",description:"The gene ontology molecular function database was downloaded from http://www.geneontology.org/. Then, we only contain the non-redundant categories by selecting the most general categories in each branch of the GO DAG structure from all categories with the number of annotated genes from 20 to 500."}],pathway:[{idtype:"entrezgene",name:"KEGG",description:"The KEGG pathway database was downloaded from http://www.kegg.jp/."},{idtype:"entrezgene",name:"Panther",description:"The PANTHER pathway database was downloaded from http://www.pantherdb.org/pathway/."},{idtype:"entrezgene",name:"Reactome",description:"The Reactome pathway database was downloaded from http://www.reactome.org/."},{idtype:"entrezgene",name:"Wikipathway",description:"The Wikipathway database was downloaded from http://www.wikipathway.org/."}],network:[{idtype:"entrezgene",name:"PPI_BIOGRID",description:"The protein-protein interaction (PPI) network was downloaded from BIOGRID (https://thebiogrid.org/). Then, we used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules."},{idtype:"entrezgene",name:"Transcription_Factor_target",description:"The relationship between transcription factor and its targets was downloaded from MsigDB (http://software.broadinstitute.org/gsea/msigdb)."},{idtype:"entrezgene",name:"miRNA_target",description:"The relationship between miRNA and its targets was downloaded from MsigDB (http://software.broadinstitute.org/gsea/msigdb)."}],disease:[],drug:[],phenotype:[],chromosomalLocation:[{idtype:"entrezgene",name:"CytogeneticBand",description:""}],"community-contributed":[],others:[]},sscrofa:{geneontology:[{idtype:"entrezgene",name:"Biological_Process",description:"The gene ontology biological process database was downloaded from http://www.geneontology.org/."},{idtype:"entrezgene",name:"Biological_Process_noRedundant",description:"The gene ontology biological process database was downloaded from http://www.geneontology.org/. Then, we only contain the non-redundant categories by selecting the most general categories in each branch of the GO DAG structure from all categories with the number of annotated genes from 20 to 500."},{idtype:"entrezgene",name:"Cellular_Component",description:"The gene ontology cellular component database was downloaded from http://www.geneontology.org/."},{idtype:"entrezgene",name:"Cellular_Component_noRedundant",description:"The gene ontology cellular component database was downloaded from http://www.geneontology.org/. Then, we only contain the non-redundant categories by selecting the most general categories in each branch of the GO DAG structure from all categories with the number of annotated genes from 20 to 500."},{idtype:"entrezgene",name:"Molecular_Function",description:"The gene ontology molecular function database was downloaded from http://www.geneontology.org/."},{idtype:"entrezgene",name:"Molecular_Function_noRedundant",description:"The gene ontology molecular function database was downloaded from http://www.geneontology.org/. Then, we only contain the non-redundant categories by selecting the most general categories in each branch of the GO DAG structure from all categories with the number of annotated genes from 20 to 500."}],pathway:[{idtype:"entrezgene",name:"KEGG",description:"The KEGG pathway database was downloaded from http://www.kegg.jp/."},{idtype:"entrezgene",name:"Wikipathway",description:"The Wikipathway database was downloaded from http://www.wikipathway.org/."}],network:[{idtype:"entrezgene",name:"CORUM",description:"The core complex subunits were downloaded from CORUM database (https://mips.helmholtz-muenchen.de/corum/#)."},{idtype:"entrezgene",name:"CORUMA",description:""},{idtype:"entrezgene",name:"PPI_BIOGRID",description:"The protein-protein interaction (PPI) network was downloaded from BIOGRID (https://thebiogrid.org/). Then, we used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules."}],disease:[],drug:[],phenotype:[],chromosomalLocation:[{idtype:"entrezgene",name:"CytogeneticBand",description:""}],"community-contributed":[],others:[]},dmelanogaster:{geneontology:[{idtype:"entrezgene",name:"Biological_Process",description:"The gene ontology biological process database was downloaded from http://www.geneontology.org/."},{idtype:"entrezgene",name:"Biological_Process_noRedundant",description:"The gene ontology biological process database was downloaded from http://www.geneontology.org/. Then, we only contain the non-redundant categories by selecting the most general categories in each branch of the GO DAG structure from all categories with the number of annotated genes from 20 to 500."},{idtype:"entrezgene",name:"Cellular_Component",description:"The gene ontology cellular component database was downloaded from http://www.geneontology.org/."},{idtype:"entrezgene",name:"Cellular_Component_noRedundant",description:"The gene ontology cellular component database was downloaded from http://www.geneontology.org/. Then, we only contain the non-redundant categories by selecting the most general categories in each branch of the GO DAG structure from all categories with the number of annotated genes from 20 to 500."},{idtype:"entrezgene",name:"Molecular_Function",description:"The gene ontology molecular function database was downloaded from http://www.geneontology.org/."},{idtype:"entrezgene",name:"Molecular_Function_noRedundant",description:"The gene ontology molecular function database was downloaded from http://www.geneontology.org/. Then, we only contain the non-redundant categories by selecting the most general categories in each branch of the GO DAG structure from all categories with the number of annotated genes from 20 to 500."}],pathway:[{idtype:"entrezgene",name:"KEGG",description:"The KEGG pathway database was downloaded from http://www.kegg.jp/."},{idtype:"entrezgene",name:"Panther",description:"The PANTHER pathway database was downloaded from http://www.pantherdb.org/pathway/."},{idtype:"entrezgene",name:"Reactome",description:"The Reactome pathway database was downloaded from http://www.reactome.org/."},{idtype:"entrezgene",name:"Wikipathway",description:"The Wikipathway database was downloaded from http://www.wikipathway.org/."}],network:[{idtype:"entrezgene",name:"PPI_BIOGRID",description:"The protein-protein interaction (PPI) network was downloaded from BIOGRID (https://thebiogrid.org/). Then, we used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules."},{idtype:"entrezgene",name:"Transcription_Factor_target",description:"The relationship between transcription factor and its targets was downloaded from MsigDB (http://software.broadinstitute.org/gsea/msigdb)."},{idtype:"entrezgene",name:"miRNA_target",description:"The relationship between miRNA and its targets was downloaded from MsigDB (http://software.broadinstitute.org/gsea/msigdb)."}],disease:[],drug:[],phenotype:[],chromosomalLocation:[{idtype:"entrezgene",name:"CytogeneticBand",description:""}],"community-contributed":[],others:[]},ggallus:{geneontology:[{idtype:"entrezgene",name:"Biological_Process",description:"The gene ontology biological process database was downloaded from http://www.geneontology.org/."},{idtype:"entrezgene",name:"Biological_Process_noRedundant",description:"The gene ontology biological process database was downloaded from http://www.geneontology.org/. Then, we only contain the non-redundant categories by selecting the most general categories in each branch of the GO DAG structure from all categories with the number of annotated genes from 20 to 500."},{idtype:"entrezgene",name:"Cellular_Component",description:"The gene ontology cellular component database was downloaded from http://www.geneontology.org/."},{idtype:"entrezgene",name:"Cellular_Component_noRedundant",description:"The gene ontology cellular component database was downloaded from http://www.geneontology.org/. Then, we only contain the non-redundant categories by selecting the most general categories in each branch of the GO DAG structure from all categories with the number of annotated genes from 20 to 500."},{idtype:"entrezgene",name:"Molecular_Function",description:"The gene ontology molecular function database was downloaded from http://www.geneontology.org/."},{idtype:"entrezgene",name:"Molecular_Function_noRedundant",description:"The gene ontology molecular function database was downloaded from http://www.geneontology.org/. Then, we only contain the non-redundant categories by selecting the most general categories in each branch of the GO DAG structure from all categories with the number of annotated genes from 20 to 500."}],pathway:[{idtype:"entrezgene",name:"KEGG",description:"The KEGG pathway database was downloaded from http://www.kegg.jp/."},{idtype:"entrezgene",name:"Panther",description:"The PANTHER pathway database was downloaded from http://www.pantherdb.org/pathway/."},{idtype:"entrezgene",name:"Reactome",description:"The Reactome pathway database was downloaded from http://www.reactome.org/."},{idtype:"entrezgene",name:"Wikipathway",description:"The Wikipathway database was downloaded from http://www.wikipathway.org/."}],network:[{idtype:"entrezgene",name:"PPI_BIOGRID",description:"The protein-protein interaction (PPI) network was downloaded from BIOGRID (https://thebiogrid.org/). Then, we used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules."},{idtype:"entrezgene",name:"Transcription_Factor_target",description:"The relationship between transcription factor and its targets was downloaded from MsigDB (http://software.broadinstitute.org/gsea/msigdb)."},{idtype:"entrezgene",name:"miRNA_target",description:"The relationship between miRNA and its targets was downloaded from MsigDB (http://software.broadinstitute.org/gsea/msigdb)."}],disease:[],drug:[],phenotype:[],chromosomalLocation:[{idtype:"entrezgene",name:"CytogeneticBand",description:""}],"community-contributed":[],others:[]},hsapiens:{geneontology:[{idtype:"entrezgene",name:"Biological_Process",description:"The gene ontology biological process database was downloaded from http://www.geneontology.org/."},{idtype:"entrezgene",name:"Biological_Process_noRedundant",description:"The gene ontology biological process database was downloaded from http://www.geneontology.org/. Then, we only contain the non-redundant categories by selecting the most general categories in each branch of the GO DAG structure from all categories with the number of annotated genes from 20 to 500."},{idtype:"entrezgene",name:"Cellular_Component",description:"The gene ontology cellular component database was downloaded from http://www.geneontology.org/."},{idtype:"entrezgene",name:"Cellular_Component_noRedundant",description:"The gene ontology cellular component database was downloaded from http://www.geneontology.org/. Then, we only contain the non-redundant categories by selecting the most general categories in each branch of the GO DAG structure from all categories with the number of annotated genes from 20 to 500."},{idtype:"entrezgene",name:"Molecular_Function",description:"The gene ontology molecular function database was downloaded from http://www.geneontology.org/."},{idtype:"entrezgene",name:"Molecular_Function_noRedundant",description:"The gene ontology molecular function database was downloaded from http://www.geneontology.org/. Then, we only contain the non-redundant categories by selecting the most general categories in each branch of the GO DAG structure from all categories with the number of annotated genes from 20 to 500."}],pathway:[{idtype:"entrezgene",name:"KEGG",description:"The KEGG pathway database was downloaded from http://www.kegg.jp/."},{idtype:"entrezgene",name:"Panther",description:"The PANTHER pathway database was downloaded from http://www.pantherdb.org/pathway/."},{idtype:"entrezgene",name:"Reactome",description:"The Reactome pathway database was downloaded from http://www.reactome.org/."},{idtype:"entrezgene",name:"Wikipathway",description:"The Wikipathway database was downloaded from http://www.wikipathway.org/."},{idtype:"entrezgene",name:"Wikipathway_cancer",description:"The Wikipathway database was downloaded from http://www.wikipathway.org/. This is the subset relevant to Clinical Proteomic Tumor Analysis Consortium."}],network:[{idtype:"entrezgene",name:"CORUM",description:"The core complex subunits were downloaded from CORUM database (https://mips.helmholtz-muenchen.de/corum/#)."},{idtype:"entrezgene",name:"CORUMA",description:""},{idtype:"entrezgene",name:"CPTAC_Proteomics_BRCA",description:"The BRCA (Breast invasive carcinoma) Proteomics data was downloaded from CPTAC (Clinical Proteomic Tumor Analysis Consortium) data portal (https://cptac-data-portal.georgetown.edu/cptacPublic/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus Proteomics co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules."},{idtype:"entrezgene",name:"CPTAC_Proteomics_COADREAD",description:"The COADREAD (Colorectal adenocarcinoma) Proteomics data was downloaded from CPTAC (Clinical Proteomic Tumor Analysis Consortium) data portal (https://cptac-data-portal.georgetown.edu/cptacPublic/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus Proteomics co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules."},{idtype:"entrezgene",name:"CPTAC_Proteomics_HNSCC",description:"The HNSCC (Head-and-neck Squamous Cell Carcinoma) Proteomics data was downloaded from LinkedOmics data portal (http://www.linkedomics.org/data_download/CPTAC-HNSCC/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus Proteomics co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules."},{idtype:"entrezgene",name:"CPTAC_Proteomics_LUAD",description:"The LUAD (Lung Adenocarcinoma) Proteomics data was downloaded from LinkedOmics data portal (http://www.linkedomics.org/data_download/CPTAC-LUAD/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus Proteomics co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules."},{idtype:"entrezgene",name:"CPTAC_Proteomics_OV",description:"The OV (Ovarian serous cystadenocarcinoma) Proteomics data was downloaded from CPTAC (Clinical Proteomic Tumor Analysis Consortium) data portal (https://cptac-data-portal.georgetown.edu/cptacPublic/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus Proteomics co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules."},{idtype:"entrezgene",name:"CPTAC_Proteomics_UCEC",description:"The UCEC (Uterine Corpus Endometrial Carcinoma) Proteomics data was downloaded from LinkedOmics data portal (http://www.linkedomics.org/data_download/CPTAC-UCEC/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus Proteomics co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules."},{idtype:"phosphositeSeq",name:"Kinase_phosphosite",description:"The relationship between Kinase and the corresponding protein phosposite position was from RegPhos (http://140.138.144.141/~RegPhos/). Then, we got the 15-mer sequence window around the phosphorylated site of the peptide (up and down 7 amino acid) from the Uniprot website (http://www.uniprot.org/) as the identifier tag."},{idtype:"phosphositeSeq",name:"Kinase_target",description:"The relationship between Kinase and its targets was from PhosphositePlus (http://www.phosphosite.org/) and the phosphorylation relationships from a in-house combined signaling network."},{idtype:"entrezgene",name:"Kinase_target",description:"The relationship between Kinase and its targets was from PhosphositePlus (http://www.phosphosite.org/) and the phosphorylation relationships from a in-house combined signaling network."},{idtype:"entrezgene",name:"PPI_BIOGRID",description:"The protein-protein interaction (PPI) network was downloaded from BIOGRID (https://thebiogrid.org/). Then, we used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules."},{idtype:"phosphositeSeq",name:"PTMsigDB",description:""},{idtype:"entrezgene",name:"TCGA_RNASeq_ACC",description:"The ACC (Adrenocortical carcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules."},{idtype:"entrezgene",name:"TCGA_RNASeq_BLCA",description:"The BLCA (Bladder Urothelial Carcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules."},{idtype:"entrezgene",name:"TCGA_RNASeq_BRCA",description:"The BRCA (Breast invasive carcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules."},{idtype:"entrezgene",name:"TCGA_RNASeq_CESC",description:"The CESC (Cervical squamous cell carcinoma and endocervical adenocarcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules."},{idtype:"entrezgene",name:"TCGA_RNASeq_CHOL",description:"The CHOL (Cholangiocarcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules."},{idtype:"entrezgene",name:"TCGA_RNASeq_COADREAD",description:"The COADREAD (Colon and rectal cancer) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules."},{idtype:"entrezgene",name:"TCGA_RNASeq_DLBC",description:"The DLBC (Lymphoid Neoplasm Diffuse Large B-cell Lymphoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules."},{idtype:"entrezgene",name:"TCGA_RNASeq_ESCA",description:"The ESCA (Esophageal carcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules."},{idtype:"entrezgene",name:"TCGA_RNASeq_GBM",description:"The GBM (Glioblastoma multiforme) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules."},{idtype:"entrezgene",name:"TCGA_RNASeq_GBMLGG",description:"The GBMLGG (Glioma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules."},{idtype:"entrezgene",name:"TCGA_RNASeq_HNSC",description:"The HNSC (Head and Neck squamous cell carcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules."},{idtype:"entrezgene",name:"TCGA_RNASeq_KICH",description:"The KICH (Kidney Chromophobe) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules."},{idtype:"entrezgene",name:"TCGA_RNASeq_KIRP",description:"The KIRP (Kidney renal papillary cell carcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules."},{idtype:"entrezgene",name:"TCGA_RNASeq_LAML",description:"The LAML (Acute Myeloid Leukemia) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules."},{idtype:"entrezgene",name:"TCGA_RNASeq_LGG",description:"The LGG (Brain Lower Grade Glioma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules."},{idtype:"entrezgene",name:"TCGA_RNASeq_LIHC",description:"The LIHC (Liver hepatocellular carcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules."},{idtype:"entrezgene",name:"TCGA_RNASeq_LUAD",description:"The LUAD (Lung adenocarcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules."},{idtype:"entrezgene",name:"TCGA_RNASeq_LUSC",description:"The LUSC (Lung squamous cell carcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules."},{idtype:"entrezgene",name:"TCGA_RNASeq_MESO",description:"The MESO (Mesothelioma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules."},{idtype:"entrezgene",name:"TCGA_RNASeq_OV",description:"The OV (Ovarian serous cystadenocarcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules."},{idtype:"entrezgene",name:"TCGA_RNASeq_PAAD",description:"The PAAD (Pancreatic adenocarcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules."},{idtype:"entrezgene",name:"TCGA_RNASeq_PCPG",description:"The PCPG (Pheochromocytoma and Paraganglioma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules."},{idtype:"entrezgene",name:"TCGA_RNASeq_PRAD",description:"The PRAD (Prostate adenocarcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules."},{idtype:"entrezgene",name:"TCGA_RNASeq_SARC",description:"The SARC (Sarcoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules."},{idtype:"entrezgene",name:"TCGA_RNASeq_SKCM",description:"The SKCM (Skin Cutaneous Melanoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules."},{idtype:"entrezgene",name:"TCGA_RNASeq_STAD",description:"The STAD (Stomach adenocarcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules."},{idtype:"entrezgene",name:"TCGA_RNASeq_STES",description:"The STES (Stomach and Esophageal carcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules."},{idtype:"entrezgene",name:"TCGA_RNASeq_TGCT",description:"The TGCT (Testicular Germ Cell Tumors) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules."},{idtype:"entrezgene",name:"TCGA_RNASeq_THCA",description:"The THCA (Thyroid carcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules."},{idtype:"entrezgene",name:"TCGA_RNASeq_THYM",description:"The THYM (Thymoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules."},{idtype:"entrezgene",name:"TCGA_RNASeq_UCEC",description:"The UCEC (Uterine Corpus Endometrial Carcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules."},{idtype:"entrezgene",name:"TCGA_RNASeq_UCS",description:"The UCS (Uterine Carcinosarcoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules."},{idtype:"entrezgene",name:"TCGA_RNASeq_UVM",description:"The UVM (Uveal Melanoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules."},{idtype:"entrezgene",name:"Transcription_Factor_target",description:"The relationship between transcription factor and its targets was downloaded from MsigDB (http://software.broadinstitute.org/gsea/msigdb)."},{idtype:"entrezgene",name:"miRNA_target",description:"The relationship between miRNA and its targets was downloaded from MsigDB (http://software.broadinstitute.org/gsea/msigdb)."}],disease:[{idtype:"entrezgene",name:"Disgenet",description:"The DisgeNET disease database was downloaded from http://www.disgenet.org."},{idtype:"entrezgene",name:"GLAD4U",description:"Disease terms were downloaded from ParmGKB (https://www.pharmgkb.org/). Genes associated with individual disease term were inferred using GLAD4U (http://glad4u.zhang-lab.org/index.php)."},{idtype:"entrezgene",name:"OMIM",description:""}],drug:[{idtype:"entrezgene",name:"DrugBank",description:"The DrugBank drug database was downloaded from https://www.drugbank.ca/."},{idtype:"entrezgene",name:"GLAD4U",description:"Drug terms were downloaded from ParmGKB (https://www.pharmgkb.org/). Genes associated with individual drug term were inferred using GLAD4U (http://glad4u.zhang-lab.org/index.php)."}],phenotype:[{idtype:"entrezgene",name:"Human_Phenotype_Ontology",description:"The phenotype ontology information for human was downloaded from http://human-phenotype-ontology.github.io/."}],chromosomalLocation:[{idtype:"entrezgene",name:"CytogeneticBand",description:""}],"community-contributed":[{idtype:"entrezgene",name:"5htGeneSets_Conte",description:""},{idtype:"entrezgene",name:"Hallmark50",description:"Hallmark gene sets summarize and represent specific well-defined biological states or processes and display coherent expression. See details at http://software.broadinstitute.org/gsea/msigdb/collections.jsp#H ."},{idtype:"entrezgene",name:"MuscleGeneSets_Duddy_2017",description:"Muscle Gene Sets (v3) Duddy 2017"}],others:[]},mmusculus:{geneontology:[{idtype:"entrezgene",name:"Biological_Process",description:"The gene ontology biological process database was downloaded from http://www.geneontology.org/."},{idtype:"entrezgene",name:"Biological_Process_noRedundant",description:"The gene ontology biological process database was downloaded from http://www.geneontology.org/. Then, we only contain the non-redundant categories by selecting the most general categories in each branch of the GO DAG structure from all categories with the number of annotated genes from 20 to 500."},{idtype:"entrezgene",name:"Cellular_Component",description:"The gene ontology cellular component database was downloaded from http://www.geneontology.org/."},{idtype:"entrezgene",name:"Cellular_Component_noRedundant",description:"The gene ontology cellular component database was downloaded from http://www.geneontology.org/. Then, we only contain the non-redundant categories by selecting the most general categories in each branch of the GO DAG structure from all categories with the number of annotated genes from 20 to 500."},{idtype:"entrezgene",name:"Molecular_Function",description:"The gene ontology molecular function database was downloaded from http://www.geneontology.org/."},{idtype:"entrezgene",name:"Molecular_Function_noRedundant",description:"The gene ontology molecular function database was downloaded from http://www.geneontology.org/. Then, we only contain the non-redundant categories by selecting the most general categories in each branch of the GO DAG structure from all categories with the number of annotated genes from 20 to 500."}],pathway:[{idtype:"entrezgene",name:"KEGG",description:"The KEGG pathway database was downloaded from http://www.kegg.jp/."},{idtype:"entrezgene",name:"Panther",description:"The PANTHER pathway database was downloaded from http://www.pantherdb.org/pathway/."},{idtype:"entrezgene",name:"Reactome",description:"The Reactome pathway database was downloaded from http://www.reactome.org/."},{idtype:"entrezgene",name:"Wikipathway",description:"The Wikipathway database was downloaded from http://www.wikipathway.org/."}],network:[{idtype:"entrezgene",name:"CORUM",description:"The core complex subunits were downloaded from CORUM database (https://mips.helmholtz-muenchen.de/corum/#)."},{idtype:"entrezgene",name:"CORUMA",description:""},{idtype:"phosphositeSeq",name:"Kinase_phosphosite",description:"The relationship between Kinase and the corresponding protein phosposite position was from RegPhos (http://140.138.144.141/~RegPhos/). Then, we got the 15-mer sequence window around the phosphorylated site of the peptide (up and down 7 amino acid) from the Uniprot website (http://www.uniprot.org/) as the identifier tag."},{idtype:"phosphositeSeq",name:"Kinase_target",description:"The relationship between Kinase and its targets was from PhosphositePlus (http://www.phosphosite.org/) and the phosphorylation relationships from a in-house combined signaling network."},{idtype:"entrezgene",name:"PPI_BIOGRID",description:"The protein-protein interaction (PPI) network was downloaded from BIOGRID (https://thebiogrid.org/). Then, we used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules."},{idtype:"phosphositeSeq",name:"PTMsigDB",description:""},{idtype:"entrezgene",name:"Transcription_Factor_target",description:"The relationship between transcription factor and its targets was downloaded from MsigDB (http://software.broadinstitute.org/gsea/msigdb)."},{idtype:"entrezgene",name:"miRNA_target",description:"The relationship between miRNA and its targets was downloaded from MsigDB (http://software.broadinstitute.org/gsea/msigdb)."}],disease:[],drug:[],phenotype:[{idtype:"entrezgene",name:"Mammalian_Phenotype_Ontology",description:"The phenotype ontology information for mouse was downloaded from http://www.informatics.jax.org/."}],chromosomalLocation:[{idtype:"entrezgene",name:"CytogeneticBand",description:""}],"community-contributed":[{idtype:"entrezgene",name:"5htGeneSets_Conte",description:""},{idtype:"entrezgene",name:"MuscleGeneSets_Duddy_2017",description:"Muscle Gene Sets (v3) Duddy 2017"}],others:[]},rnorvegicus:{geneontology:[{idtype:"entrezgene",name:"Biological_Process",description:"The gene ontology biological process database was downloaded from http://www.geneontology.org/."},{idtype:"entrezgene",name:"Biological_Process_noRedundant",description:"The gene ontology biological process database was downloaded from http://www.geneontology.org/. Then, we only contain the non-redundant categories by selecting the most general categories in each branch of the GO DAG structure from all categories with the number of annotated genes from 20 to 500."},{idtype:"entrezgene",name:"Cellular_Component",description:"The gene ontology cellular component database was downloaded from http://www.geneontology.org/."},{idtype:"entrezgene",name:"Cellular_Component_noRedundant",description:"The gene ontology cellular component database was downloaded from http://www.geneontology.org/. Then, we only contain the non-redundant categories by selecting the most general categories in each branch of the GO DAG structure from all categories with the number of annotated genes from 20 to 500."},{idtype:"entrezgene",name:"Molecular_Function",description:"The gene ontology molecular function database was downloaded from http://www.geneontology.org/."},{idtype:"entrezgene",name:"Molecular_Function_noRedundant",description:"The gene ontology molecular function database was downloaded from http://www.geneontology.org/. Then, we only contain the non-redundant categories by selecting the most general categories in each branch of the GO DAG structure from all categories with the number of annotated genes from 20 to 500."}],pathway:[{idtype:"entrezgene",name:"KEGG",description:"The KEGG pathway database was downloaded from http://www.kegg.jp/."},{idtype:"entrezgene",name:"Panther",description:"The PANTHER pathway database was downloaded from http://www.pantherdb.org/pathway/."},{idtype:"entrezgene",name:"Reactome",description:"The Reactome pathway database was downloaded from http://www.reactome.org/."},{idtype:"entrezgene",name:"Wikipathway",description:"The Wikipathway database was downloaded from http://www.wikipathway.org/."}],network:[{idtype:"entrezgene",name:"CORUM",description:"The core complex subunits were downloaded from CORUM database (https://mips.helmholtz-muenchen.de/corum/#)."},{idtype:"entrezgene",name:"CORUMA",description:""},{idtype:"phosphositeSeq",name:"Kinase_phosphosite",description:"The relationship between Kinase and the corresponding protein phosposite position was from RegPhos (http://140.138.144.141/~RegPhos/). Then, we got the 15-mer sequence window around the phosphorylated site of the peptide (up and down 7 amino acid) from the Uniprot website (http://www.uniprot.org/) as the identifier tag."},{idtype:"phosphositeSeq",name:"Kinase_target",description:"The relationship between Kinase and its targets was from PhosphositePlus (http://www.phosphosite.org/) and the phosphorylation relationships from a in-house combined signaling network."},{idtype:"entrezgene",name:"PPI_BIOGRID",description:"The protein-protein interaction (PPI) network was downloaded from BIOGRID (https://thebiogrid.org/). Then, we used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules."},{idtype:"phosphositeSeq",name:"PTMsigDB",description:""},{idtype:"entrezgene",name:"Transcription_Factor_target",description:"The relationship between transcription factor and its targets was downloaded from MsigDB (http://software.broadinstitute.org/gsea/msigdb)."},{idtype:"entrezgene",name:"miRNA_target",description:"The relationship between miRNA and its targets was downloaded from MsigDB (http://software.broadinstitute.org/gsea/msigdb)."}],disease:[],drug:[],phenotype:[],chromosomalLocation:[{idtype:"entrezgene",name:"CytogeneticBand",description:""}],"community-contributed":[],others:[]},scerevisiae:{geneontology:[{idtype:"entrezgene",name:"Biological_Process",description:"The gene ontology biological process database was downloaded from http://www.geneontology.org/."},{idtype:"entrezgene",name:"Biological_Process_noRedundant",description:"The gene ontology biological process database was downloaded from http://www.geneontology.org/. Then, we only contain the non-redundant categories by selecting the most general categories in each branch of the GO DAG structure from all categories with the number of annotated genes from 20 to 500."},{idtype:"entrezgene",name:"Cellular_Component",description:"The gene ontology cellular component database was downloaded from http://www.geneontology.org/."},{idtype:"entrezgene",name:"Cellular_Component_noRedundant",description:"The gene ontology cellular component database was downloaded from http://www.geneontology.org/. Then, we only contain the non-redundant categories by selecting the most general categories in each branch of the GO DAG structure from all categories with the number of annotated genes from 20 to 500."},{idtype:"entrezgene",name:"Molecular_Function",description:"The gene ontology molecular function database was downloaded from http://www.geneontology.org/."},{idtype:"entrezgene",name:"Molecular_Function_noRedundant",description:"The gene ontology molecular function database was downloaded from http://www.geneontology.org/. Then, we only contain the non-redundant categories by selecting the most general categories in each branch of the GO DAG structure from all categories with the number of annotated genes from 20 to 500."}],pathway:[{idtype:"entrezgene",name:"KEGG",description:"The KEGG pathway database was downloaded from http://www.kegg.jp/."},{idtype:"entrezgene",name:"Panther",description:"The PANTHER pathway database was downloaded from http://www.pantherdb.org/pathway/."},{idtype:"entrezgene",name:"Reactome",description:"The Reactome pathway database was downloaded from http://www.reactome.org/."},{idtype:"entrezgene",name:"Wikipathway",description:"The Wikipathway database was downloaded from http://www.wikipathway.org/."}],network:[{idtype:"entrezgene",name:"PPI_BIOGRID",description:"The protein-protein interaction (PPI) network was downloaded from BIOGRID (https://thebiogrid.org/). Then, we used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules."},{idtype:"entrezgene",name:"Transcription_Factor_target",description:"The relationship between transcription factor and its targets was downloaded from MsigDB (http://software.broadinstitute.org/gsea/msigdb)."},{idtype:"entrezgene",name:"miRNA_target",description:"The relationship between miRNA and its targets was downloaded from MsigDB (http://software.broadinstitute.org/gsea/msigdb)."}],disease:[],drug:[],phenotype:[],chromosomalLocation:[{idtype:"entrezgene",name:"CytogeneticBand",description:""}],"community-contributed":[],others:[]},others:[]}},42:function(e){e.exports={hsapiens:"Homo sapiens",athaliana:"Arabidopsis thaliana",btaurus:"Bos taurus",celegans:"Caenorhabditis elegans",cfamiliaris:"Canis lupus familiaris",dmelanogaster:"Drosophila melanogaster",drerio:"Danio rerio",ggallus:"Gallus gallus",mmusculus:"Mus musculus",rnorvegicus:"Rattus norvegicus",scerevisiae:"Saccharomyces cerevisiae",sscrofa:"Sus scrofa"}},43:function(e){e.exports={phosphositeUniprot:"Uniprot protein ID with phosphosite annotation",phosphositeRefseq:"Refseq protein ID with phosphosite annotation",phosphositeEnsembl:"Ensembl protein ID with phosphosite annotation",phosphositeSeq:"Phosphosite sequence",uniprotswissprot:"Uniprot Swiss-Prot",genename:"Gene name",entrezgene:"NCBI Entrez gene",genesymbol:"Gene symbol"}},44:function(e,t,o){"use strict";var n=o(23);o.n(n).a},45:function(e,t,o){},48:function(e,t,o){"use strict";o.r(t);var n=o(5),a=o.n(n),i=o(41),s=o(42),r=o(43),d={props:["initialValues"],data:function(){return{advOpen:!1,organisms:Object.keys(i),organism:this.initialValues.organism||"hsapiens",enrichMethod:this.initialValues.enrichMethod||"ORA",databases:this.initialValues.databases||[{category:null,name:null}],databaseFile:null,databaseDesFile:null,databaseFileIdType:null,dbType:null,standardId:null,networkData:null,idData:null,idType:this.initialValues.idType||"genesymbol",geneListFile:null,geneList:this.initialValues.geneList||null,refData:null,reference:this.initialValues.reference||null,referenceFile:null,refFileIdType:null,minNum:this.initialValues.minNum||5,maxNum:this.initialValues.maxNum||2e3,fdrMethod:this.initialValues.fdrMethod||"BH",sigMethod:this.initialValues.sigMethod||"top",sigValue:this.initialValues.sigValue||10,setCoverNum:this.initialValues.setCoverNum||10,reportNum:this.initialValues.reportNum||40,colorScheme:this.initialValues.colorScheme||"continuous",perNum:this.initialValues.perNum||1e3,gseaP:this.initialValues.gseaP||1,collapseMethod:this.initialValues.collapseMethod||"mean",ntaMethod:this.initialValues.ntaMethod||"Network_Expansion",neighborNum:this.initialValues.neighborNum||10,highlightSeedNum:this.initialValues.highlightSeedNum||10,ntaHighlight:this.initialValues.ntaHighlight||"Seeds",errorGeneList:!1,msgGeneList:"",errorGeneListFile:!1,msgGeneListFIle:""}},computed:{categoryOptions:function(){var t=this;return"NTA"==this.enrichMethod?["network"]:"others"==this.organism?[]:Object.keys(i[this.organism]).filter(function(e){return i[t.organism][e].length})},databaseOptions:function(){var t=this,n=this.databases.map(function(e){return e.name});return this.databases.map(function(e,o){return"others"==t.organism||null===e.category?[{name:null,description:null,idtype:null,label:null}]:"NTA"!=t.enrichMethod?i[t.organism][e.category].filter(function(e){return"GSEA"!=t.enrichMethod||"geneontology"!=t.databases[0].category||/.*_noRedundant/.test(e.name)}).filter(function(t){return n.indexOf(t.name)===o||!n.map(function(e){return e&&e.includes(t.name)}).includes(!0)&&!n.map(function(e){return t.name&&t.name.includes(e)}).includes(!0)}).map(function(e){return e.label=e.name.replace(/_/g," "),e}):null===t.networkData?null:t.networkData[t.organism][e.category].map(function(e){return e.label=e.name.replace(/_/g," "),e})})},idTypes:function(){var t=this;if(null===this.idData||"others"===this.organism)return null;var o=Object.keys(r);return this.idData[this.organism].filter(function(e){return(null===t.standardId||e.type==t.standardId)&&"entrezgene_protein-coding"!=e.name}).sort(function(e,t){return o.indexOf(t.name)-o.indexOf(e.name)}).map(function(e){return{name:e.name,example:e.E5+"\n"+e.E1+"\n"+e.E2+"\n"+e.E3+"\n"+e.E4,label:e.name in r?r[e.name]:e.name.replace(/_/g," ")}})},references:function(){var t=this;return null===this.refData||"others"==this.organism?null:this.refData[this.organism].filter(function(e){return null===t.standardId||e.type==t.standardId}).map(function(e){return{name:e.name,label:e.name.replace(/_/g," ")}})},isValid:function(){var e,t=!this.errorGeneList&&!this.errorGeneListFile,o="fdr"===this.sigMethod&&0<=this.sigValue&&this.sigValue<=1||"top"===this.sigMethod&&0<this.sigValue;return"ORA"==this.enrichMethod?e=!(!this.databases&&!this.databaseFile||"others"!=this.organism&&!this.idType||!this.geneList&&!this.geneListFile||!this.reference&&(!this.referenceFile||"others"!=this.organism&&!this.refFileIdType)):"GSEA"==this.enrichMethod?e=!(!this.databases&&!this.databaseFile||"others"!=this.organism&&!this.idType||!this.geneList&&!this.geneListFile):"NTA"==this.enrichMethod&&(e=!(!this.databases||"others"!=this.organism&&!this.idType||!this.geneList&&!this.geneListFile)),t&&o&&e}},methods:{addDatbase:function(){this.databases.push({category:null,name:null})},removeDatbase:function(e){this.databases.splice(e,1)},setStandardId:function(){var t=this.databases[0];if(null===t.name||"others"==this.organism)return this.dbType=null,void(this.standardId=null);var e=i[this.organism][t.category].find(function(e){return e.name==t.name});this.standardId=e.idtype,"entrezgene"==e.idtype?this.dbType="gene":this.dbType="phosphosite"},setStandardIdByIdtype:function(){var t=this;this.idData&&"others"!=this.organism&&(this.standardId=this.idData[this.organism].find(function(e){return e.name==t.databaseFileIdType}).type)},setDbIdType:function(){this.databaseFile=null,this.databaseDesFile=null,this.databaseFileIdType=null,this.setStandardId()},resetRefSection:function(){this.referenceFile=null,this.refFileIdType=null,this.reference=null},onSigChange:function(){var e=this;"fdr"!=this.sigMethod||this.isValid?"top"!=this.sigMethod||this.isValid&&this.sigValue===Math.floor(this.sigValue)||(this.sigValue=10):this.sigValue=.05,this.$nextTick(function(){e.$refs.sigValue.checkHtml5Validity()})},validate:function(){var t=this;if(this.errorGeneList=!1,this.msgGeneList="",this.errorGeneListFile=!1,this.msgGeneListFile="","others"!=this.organism&&(this.geneList||this.geneListFile)){var e=new FormData;e.append("organism",this.organism),e.append("enrich_method",this.enrichMethod),e.append("databases",JSON.stringify(this.databases)),e.append("ref_set",this.reference),e.append("id_type",this.idType),e.append("gene_list",this.geneList),e.append("standard_id",this.standardId),e.append("gene_list_file",this.geneListFile),fetch("./api/validation",{method:"POST",body:e}).then(function(e){return e.json()}).then(function(e){"status"in e&&1==e.status&&("gene_list"==e.element?(t.errorGeneList=!0,t.msgGeneList=e.message):"gene_list_file"==e.element&&(t.errorGeneListFile=!0,t.msgGeneListFile=e.message))})}}},watch:{initialValues:function(){var e=this;this.databaseFile=null,this.databaseDesFile=null,this.databaseFileIdType=null,this.geneListFile=null,this.reference=null,this.referenceFile=null,this.refFileIdType=null,this.errorGeneList=!1,this.msgGeneList="",this.errorGeneListFile=!1,this.msgGeneListFile="";var t=JSON.parse(JSON.stringify(this.initialValues));"organism"in t&&(this.organism=t.organism),"enrichMethod"in t&&(this.enrichMethod=t.enrichMethod),this.$nextTick(function(){"databases"in t&&(e.databases=t.databases)}),"idType"in t&&(this.idType=t.idType),"geneList"in t&&(this.geneList=t.geneList),"reference"in t&&(this.reference=t.reference),"minNum"in t&&(this.minNum=t.minNum),"maxNum"in t&&(this.maxNum=t.maxNum),"fdrMethod"in t&&(this.fdrMethod=t.fdrMethod),"sigMethod"in t&&(this.sigMethod=t.sigMethod),"sigValue"in t&&(this.sigValue=t.sigValue),"reportNum"in t&&(this.reportNum=t.reportNum),"perNum"in t&&(this.perNum=t.perNum),"gseaP"in t&&(this.gseaP=t.gseaP),"collapseMethod"in t&&(this.collapseMethod=t.collapseMethod),"ntaMethod"in t&&(this.ntaMethod=t.ntaMethod),"higlightSeedNum"in t&&(this.higlightSeedNum=t.higlightSeedNum),"neighborNum"in t&&(this.neighborNum=t.neighborNum),"ntaHighlight"in t&&(this.ntaHighlight=t.ntaHighlight)},organism:function(e){this.databases="others"==e?(this.idType=null,[{category:"others",name:""}]):[{category:null,name:null}]},enrichMethod:function(e){var t=this;this.databases=[{category:null,name:null}],"NTA"!=e||this.networkData||fetch("./data/networksummary.json").then(function(e){return e.json()}).then(function(e){return t.networkData=e})},databases:function(){this.setStandardId()}},created:function(){this.organismNames=s,this.organismNames.others="others"},mounted:function(){var t=this;fetch("./data/idtypesummary.json").then(function(e){return e.json()}).then(function(e){return t.idData=e}),fetch("./data/referencesetsummary.json").then(function(e){return e.json()}).then(function(e){return t.refData=e})}},l=(o(44),o(1)),c=Object(l.a)(d,function(){var n=this,e=n.$createElement,a=n._self._c||e;return a("form",{attrs:{id:"wgForm",target:"_blank",action:"./process.php",method:"POST",enctype:"multipart/form-data"}},[a("fieldset",{staticClass:"param basic"},[a("legend",[n._v("Basic parameters")]),n._v(" "),a("div",{staticClass:"field is-horizontal"},[a("div",{staticClass:"field-label"},[a("label",{staticClass:"label"},[n._v("Organism of Interest\n          "),a("b-tooltip",{attrs:{label:"Select an organism from the dropdown menu. If selecting 'others', users can upload functional databases or experimental data from other organisms or with gene identifiers not covered by WebGestalt.",type:"is-light",multilined:""}},[a("b-icon",{attrs:{icon:"help-circle-outline",size:"is-small"}})],1)],1)]),n._v(" "),a("div",{staticClass:"field-body"},[a("div",{staticClass:"field"},[a("div",{staticClass:"control is-expanded"},[a("b-select",{attrs:{id:"organism",name:"organism",size:"is-small",expanded:""},model:{value:n.organism,callback:function(e){n.organism=e},expression:"organism"}},n._l(n.organisms,function(e){return a("option",{key:e,domProps:{value:e}},[n._v("\n                "+n._s(n.organismNames[e])+"\n              ")])}),0)],1)])])]),n._v(" "),a("div",{staticClass:"field is-horizontal"},[a("div",{staticClass:"field-label"},[a("label",{staticClass:"label"},[n._v("Method of Interest\n          "),a("b-tooltip",{attrs:{label:"Select a method from the dropdown menu for enrichment analysis. If selecting 'Network Topology-based Analysis (NTA)', WebGestalt will integrate our web tool Gene2Net toperform network-based enrichment analysis. Detailed information about Gene2Net can be found at www.gene2net.org.",type:"is-light",multilined:""}},[a("b-icon",{attrs:{icon:"help-circle-outline",size:"is-small"}})],1)],1)]),n._v(" "),a("div",{staticClass:"field-body"},[a("div",{staticClass:"field"},[a("div",{staticClass:"control is-expanded"},[a("b-select",{attrs:{name:"enrich_method",size:"is-small",expanded:""},model:{value:n.enrichMethod,callback:function(e){n.enrichMethod=e},expression:"enrichMethod"}},[a("option",{attrs:{value:"ORA"}},[n._v("Over-Representation Analysis (ORA)")]),n._v(" "),a("option",{attrs:{value:"GSEA"}},[n._v("Gene Set Enrichment Analysis(GSEA)")]),n._v(" "),a("option",{attrs:{value:"NTA"}},[n._v("Network Topology-based Analysis (NTA)")])])],1)])])]),n._v(" "),n._l(n.databases,function(t,o){return[a("div",{staticClass:"field is-horizontal"},[a("div",{staticClass:"field-label"},[a("label",{staticClass:"label"},[n._v("Funtional Database\n          "),a("b-tooltip",{attrs:{label:"Select a functional database category and a database in the category from the two dropdown menus. If selecting 'others' option in category dropdown menu, users can upload their own functional database for the selected organism. Currently, only one custom database is supported.",type:"is-light",multilined:""}},[a("b-icon",{attrs:{icon:"help-circle-outline",size:"is-small"}})],1)],1)]),n._v(" "),a("div",{staticClass:"field-body"},[a("div",{staticClass:"field"},[a("div",{staticClass:"control is-expanded"},[a("b-select",{attrs:{name:"enriched_database_category[]",size:"is-small",placeholder:"Select a function database category",required:"",expanded:""},on:{input:function(){t.name=null,0===o&&n.setDbIdType()}},model:{value:t.category,callback:function(e){n.$set(t,"category",e)},expression:"db.category"}},[n._l(n.categoryOptions,function(e){return"others"!==e||0===o?a("option",{key:e,domProps:{value:e}},[n._v("\n                "+n._s(e)+"\n              ")]):n._e()}),n._v(" "),0===o&&"NTA"!=n.enrichMethod?a("option",{key:"others",attrs:{value:"others"}},[n._v("others")]):n._e()],2)],1)])])]),n._v(" "),a("div",{staticClass:"field is-horizontal"},[a("div",{staticClass:"field-label"},[a("label",{staticClass:"label"},["ORA"==n.enrichMethod&&"others"!=n.organism?[0==o?a("a",{staticClass:"button is-small",on:{click:n.addDatbase}},[a("b-tooltip",{attrs:{label:"Add database combination for search",type:"is-light"}},[a("b-icon",{attrs:{icon:"plus"}})],1)],1):a("a",{staticClass:"button is-small",on:{click:function(e){return n.removeDatbase(o)}}},[a("b-icon",{attrs:{icon:"minus"}})],1)]:n._e()],2)]),n._v(" "),a("div",{staticClass:"field-body"},[a("div",{staticClass:"field"},[a("div",{staticClass:"control is-expanded"},[a("b-select",{attrs:{name:"enriched_database_name[]",size:"is-small",placeholder:"Select a function database name",disabled:null===t.name&&null===t.category||"others"===t.category,required:"",expanded:""},on:{input:function(){0===o&&n.setDbIdType(),n.validate()}},model:{value:t.name,callback:function(e){n.$set(t,"name",e)},expression:"db.name"}},n._l(n.databaseOptions[o],function(e){return a("option",{key:e.name,attrs:{title:e.description,disabled:0<o&&n.standardId&&e.idtype!==n.standardId},domProps:{value:e.name}},[n._v("\n                "+n._s(e.label)+"\n              ")])}),0)],1)])])]),n._v(" "),"others"===n.databases[o].category?[a("div",{staticClass:"field is-horizontal"},[n._m(0,!0),n._v(" "),a("div",{staticClass:"field-body"},[a("div",{staticClass:"field"},[a("div",{staticClass:"control is-expanded"},[a("b-select",{attrs:{name:"uploaded_db_id_type",size:"is-small",placeholder:"Select ID type of the custom database",expanded:"",disabled:null===n.idTypes},on:{input:n.setStandardIdByIdtype},model:{value:n.databaseFileIdType,callback:function(e){n.databaseFileIdType=e},expression:"databaseFileIdType"}},n._l(n.idTypes,function(e){return a("option",{key:e.name,attrs:{title:e.example},domProps:{value:e.name}},[n._v("\n                  "+n._s(e.label)+"\n                ")])}),0)],1)])])]),n._v(" "),a("div",{staticClass:"field is-horizontal"},[a("div",{staticClass:"field-label is-small"},[a("label",{staticClass:"label"},[n._v("Upload Functional Database\n            "),a("b-tooltip",{attrs:{label:"Upload a functional database with gmt extension.",type:"is-light",multilined:""}},[a("b-icon",{attrs:{icon:"help-circle-outline",size:"is-small"}})],1)],1)]),n._v(" "),a("div",{staticClass:"field-body"},[a("b-field",{staticClass:"file"},[a("b-upload",{attrs:{name:"upload_enriched_databasefile",accept:".gmt",native:""},model:{value:n.databaseFile,callback:function(e){n.databaseFile=e},expression:"databaseFile"}},[a("a",{staticClass:"button is-outlined is-small"},[a("b-icon",{attrs:{icon:"upload",size:"is-small"}}),a("span",[n._v("Click to upload")])],1)]),n._v(" "),n.databaseFile?a("span",{staticClass:"file-name"},[n._v(n._s(n.databaseFile.name))]):n._e()],1)],1)]),n._v(" "),a("div",{staticClass:"field is-horizontal"},[a("div",{staticClass:"field-label is-small"},[a("label",{staticClass:"label"},[n._v("Upload Database Description File (Optional)\n            "),a("b-tooltip",{attrs:{label:"Upload database description file with des extension. The category IDs in the description file should be the exactly same with those in the uploaded gmt file.",type:"is-light",multilined:""}},[a("b-icon",{attrs:{icon:"help-circle-outline",size:"is-small"}})],1)],1)]),n._v(" "),a("div",{staticClass:"field-body"},[a("b-field",{staticClass:"file"},[a("b-upload",{attrs:{name:"upload_enriched_database_descriptionfile",accept:".des",native:""},model:{value:n.databaseDesFile,callback:function(e){n.databaseDesFile=e},expression:"databaseDesFile"}},[a("a",{staticClass:"button is-outlined is-small"},[a("b-icon",{attrs:{icon:"upload",size:"is-small"}}),a("span",[n._v("Click to upload")])],1)]),n._v(" "),n.databaseDesFile?a("span",{staticClass:"file-name"},[n._v(n._s(n.databaseDesFile.name))]):n._e()],1)],1)])]:n._e()]}),n._v(" "),n._m(1),n._v(" "),a("div",{staticClass:"field is-horizontal"},[a("div",{staticClass:"field-label"},[a("label",{staticClass:"label"},[n._v("Select Gene ID Type\n          "),a("b-tooltip",{attrs:{label:"Select an ID Type from the dropdown menu that corresponds to the uploaded gene list. When hovering on one option, some id type examples for this option will be shown.",type:"is-light",multilined:""}},[a("b-icon",{attrs:{icon:"help-circle-outline",size:"is-small"}})],1)],1)]),n._v(" "),a("div",{staticClass:"field-body"},[a("div",{staticClass:"field"},[a("div",{staticClass:"control is-expanded"},[a("b-select",{attrs:{name:"id_type",size:"is-small",placeholder:"Select the ID type of input genes",expanded:"",disabled:null===n.idTypes},on:{input:n.validate},model:{value:n.idType,callback:function(e){n.idType=e},expression:"idType"}},n._l(n.idTypes,function(e){return a("option",{key:e.name,attrs:{title:e.example},domProps:{value:e.name}},[n._v("\n                "+n._s(e.label)+"\n              ")])}),0)],1)])])]),n._v(" "),a("div",{staticClass:"field is-horizontal"},[a("div",{staticClass:"field-label"},[a("label",{staticClass:"label"},[n._v("Upload Gene List\n          "),a("b-tooltip",{attrs:{label:"Upload the interesting gene list. When selecting ORA method, users need to upload a txt file with only one column or paste a gene list to the text box. If selecting the GSEA method, users should upload a ranked list file (rnk file) with two columns: gene IDs and the scores separated by tab or paste gene IDs and the scores separated by tab in the text box. Maximum upload file size is 5MB.",type:"is-light",multilined:""}},[a("b-icon",{attrs:{icon:"help-circle-outline",size:"is-small"}})],1)],1)]),n._v(" "),a("div",{staticClass:"field-body"},[a("div",{staticClass:"msg-wrapper"},[a("b-field",{attrs:{grouped:""}},[a("b-field",{staticClass:"file"},[a("b-upload",{attrs:{name:"gene_list_file",accept:"GSEA"==n.enrichMethod?".rnk":".txt",disabled:!!n.geneList,native:""},on:{input:n.validate},model:{value:n.geneListFile,callback:function(e){n.geneListFile=e},expression:"geneListFile"}},[a("a",{staticClass:"button is-outlined is-small",class:{"is-danger":n.errorGeneListFile},attrs:{disabled:!!n.geneList}},[a("b-icon",{attrs:{icon:"upload",size:"is-small"}}),a("span",[n._v("Click to upload")])],1)]),n._v(" "),n.geneListFile?a("span",{staticClass:"file-name"},[n._v(n._s(n.geneListFile.name))]):n._e()],1),n._v(" "),a("a",{staticClass:"button is-small",on:{click:function(){n.geneListFile=null,n.geneList=null}}},[n._v("Reset")])],1),n._v(" "),n.errorGeneListFile?a("p",{staticClass:"upload-gene-list-msg help is-danger"},[n._v(n._s(n.msgGeneListFile))]):n._e()],1)])]),n._v(" "),a("div",{staticClass:"field is-horizontal"},[n._m(2),n._v(" "),a("div",{staticClass:"field-body"},[a("div",{staticClass:"gene-list-wrapper"},[a("p",{staticClass:"or-spacer has-text-primary"},[n._v("OR")]),n._v(" "),a("b-field",{attrs:{type:{"is-danger":n.errorGeneList},message:n.msgGeneList}},[a("div",{staticStyle:{position:"relative"}},[a("a",{staticClass:"delete is-small input-clear",on:{click:function(e){n.geneList=null}}}),n._v(" "),a("b-input",{attrs:{type:"textarea",placeholder:"Please enter gene ids...",name:"gene_list",disabled:!!n.geneListFile},on:{blur:n.validate},model:{value:n.geneList,callback:function(e){n.geneList=e},expression:"geneList"}})],1)])],1)])]),n._v(" "),"ORA"===n.enrichMethod?[n._m(3),n._v(" "),a("div",{staticClass:"field is-horizontal"},[a("div",{staticClass:"field-label"},[a("label",{staticClass:"label"},[n._v("Select Reference Set\n          "),a("b-tooltip",{attrs:{label:"Select a built-in reference set or microarray platform for ORA.",type:"is-light",multilined:""}},[a("b-icon",{attrs:{icon:"help-circle-outline",size:"is-small"}})],1)],1)]),n._v(" "),a("div",{staticClass:"field-body"},[a("div",{staticClass:"field"},[a("div",{staticClass:"control is-expanded"},[a("b-select",{attrs:{name:"ref_set",size:"is-small",placeholder:"Select the reference set",disabled:!!n.referenceFile||null===n.references,expanded:""},model:{value:n.reference,callback:function(e){n.reference=e},expression:"reference"}},n._l(n.references,function(e){return a("option",{key:e.name,domProps:{value:e.name}},[n._v("\n                "+n._s(e.label)+"\n              ")])}),0)],1)])])]),n._v(" "),a("div",{staticClass:"field is-horizontal"},[a("div",{staticClass:"field-label"},[a("label",{staticClass:"label"},[n._v("Upload User Reference Set File and Select ID type\n          "),a("b-tooltip",{attrs:{label:"Upload the reference gene list in a txt file with one colum.",type:"is-light",multilined:""}},[a("b-icon",{attrs:{icon:"help-circle-outline",size:"is-small"}})],1)],1)]),n._v(" "),a("div",{staticClass:"field-body"},[a("div",[a("div",{staticClass:"field",attrs:{id:"ref-id-type"}},[a("div",{staticClass:"control is-expanded"},[a("b-select",{attrs:{name:"ref_id_type",size:"is-small",placeholder:"Select the ID type of reference set",expanded:"",disabled:!!n.reference||null===n.idTypes},model:{value:n.refFileIdType,callback:function(e){n.refFileIdType=e},expression:"refFileIdType"}},n._l(n.idTypes,function(e){return a("option",{key:e.name,attrs:{title:e.example},domProps:{value:e.name}},[n._v("\n                  "+n._s(e.label)+"\n                ")])}),0)],1)]),n._v(" "),a("b-field",{attrs:{grouped:""}},[a("b-field",{staticClass:"file"},[a("b-upload",{attrs:{name:"ref_file",accept:".txt",disabled:!!n.reference,native:""},model:{value:n.referenceFile,callback:function(e){n.referenceFile=e},expression:"referenceFile"}},[a("a",{staticClass:"button is-outlined is-small",attrs:{disabled:!!n.reference}},[a("b-icon",{attrs:{icon:"upload",size:"is-small"}}),a("span",[n._v("Click to upload")])],1)]),n._v(" "),n.referenceFile?a("span",{staticClass:"file-name"},[n._v(n._s(n.referenceFile.name))]):n._e()],1),n._v(" "),a("a",{staticClass:"button is-small",on:{click:n.resetRefSection}},[n._v("Reset")])],1)],1)])])]:n._e()],2),n._v(" "),a("fieldset",[a("legend",{staticClass:"adv-legend",on:{click:function(e){n.advOpen=!n.advOpen}}},[n._v("Advanced parameters\n      "),a("b-icon",{attrs:{size:"is-small",icon:n.advOpen?"chevron-down":"chevron-up"}})],1),n._v(" "),a("div",{directives:[{name:"show",rawName:"v-show",value:n.advOpen,expression:"advOpen"}]},["NTA"!=n.enrichMethod?[a("div",{staticClass:"field is-horizontal"},[a("div",{staticClass:"field-label"},[a("label",{staticClass:"label"},[n._v("minimum number of genes for a category\n            "),a("b-tooltip",{attrs:{label:"the category with a size less than this number will be removed during the analysis. the category size is calculated based on the number of overlapping genes between the annotated genes in the category and the reference gene list for the ora method (or ranked gene list for the gsea method).",type:"is-light",multilined:""}},[a("b-icon",{attrs:{icon:"help-circle-outline",size:"is-small"}})],1)],1)]),n._v(" "),a("div",{staticClass:"field-body"},[a("b-input",{attrs:{name:"min_num",size:"is-small",type:"number",min:"1"},model:{value:n.minNum,callback:function(e){n.minNum=e},expression:"minNum"}})],1)]),n._v(" "),a("div",{staticClass:"field is-horizontal"},[a("div",{staticClass:"field-label"},[a("label",{staticClass:"label"},[n._v("Maximum number of genes for a category\n            "),a("b-tooltip",{attrs:{label:"The category with a size greater than this number will be removed during the analysis. The category size is calculated based on the number of overlapping genes between the annotated genes in the category and the reference gene list for the ORA method (or ranked gene list for the GSEA method).",type:"is-light",multilined:""}},[a("b-icon",{attrs:{icon:"help-circle-outline",size:"is-small"}})],1)],1)]),n._v(" "),a("div",{staticClass:"field-body"},[a("b-input",{attrs:{name:"max_num",size:"is-small",type:"number",min:"1"},model:{value:n.maxNum,callback:function(e){n.maxNum=e},expression:"maxNum"}})],1)])]:n._e(),n._v(" "),"ORA"==n.enrichMethod?[a("div",{staticClass:"field is-horizontal"},[a("div",{staticClass:"field-label"},[a("label",{staticClass:"label"},[n._v("Multiple Test Adjustment\n            "),a("b-tooltip",{attrs:{label:"As we are testing multiple gene sets at the same time, the p values need to be adjusted. We use the R function p.adjust for this purpose. Information on this function is available at http://rweb.stat.umn.edu/R/library/stats/html/p.adjust.html",type:"is-light",multilined:""}},[a("b-icon",{attrs:{icon:"help-circle-outline",size:"is-small"}})],1)],1)]),n._v(" "),a("div",{staticClass:"field-body"},[a("b-select",{attrs:{name:"fdr_method",size:"is-small"},model:{value:n.fdrMethod,callback:function(e){n.fdrMethod=e},expression:"fdrMethod"}},[a("option",{attrs:{value:"BH"}},[n._v("BH")]),n._v(" "),a("option",{attrs:{value:"BY"}},[n._v("BY")]),n._v(" "),a("option",{attrs:{value:"bonferroni"}},[n._v("Bonferroni")]),n._v(" "),a("option",{attrs:{value:"holm"}},[n._v("Holm")]),n._v(" "),a("option",{attrs:{value:"hommel"}},[n._v("Hommel")])])],1)])]:n._e(),n._v(" "),"NTA"==n.enrichMethod?[a("div",{staticClass:"field is-horizontal"},[a("div",{staticClass:"field-label"},[a("label",{staticClass:"label"},[n._v("Network Construction Method\n              "),a("b-tooltip",{attrs:{label:"Two methods are available for network construction: Network Retrieval & Prioritization first uses random walk analysis to calculate random walk probability for the input seeds, then identifies the relationships among the seeds in the selected network and returns a retrieval sub-network. The seeds with the top random walk probability are highlighted in the sub-network. Network Expansion first uses random walk analysis to rank all genes in the selected network based on their network proximity to the input seeds and then return an expanded sub-network in which nodes are the input seeds and their top ranking neighbors and edges represent their relationships.",type:"is-light",multilined:""}},[a("b-icon",{attrs:{icon:"help-circle-outline",size:"is-small"}})],1)],1)]),n._v(" "),a("div",{staticClass:"field-body"},[a("b-select",{attrs:{name:"NTA_method",size:"is-small"},model:{value:n.ntaMethod,callback:function(e){n.ntaMethod=e},expression:"ntaMethod"}},[a("option",{attrs:{value:"Network_Expansion"}},[n._v("Network Expansion")]),n._v(" "),a("option",{attrs:{value:"Network_Retrieval_Prioritization"}},[n._v("Network Retrieval & Prioritization")])])],1)]),n._v(" "),"Network_Expansion"==n.ntaMethod?[a("div",{staticClass:"field is-horizontal"},[a("div",{staticClass:"field-label"},[a("label",{staticClass:"label"},[n._v("Set Number of Top Ranking Neighbors\n                "),a("b-tooltip",{attrs:{label:"Based on the random walk analysis and the input seeds, each node in the network will have a score that represents the steady-state probability that the random walker would stay at the node. Then, we remove the input seeds and rank the scores of other genes from high to low and select the top ranking genes to output. The final network will contain input seeds and top ranking neighbors.",type:"is-light",multilined:""}},[a("b-icon",{attrs:{icon:"help-circle-outline",size:"is-small"}})],1)],1)]),n._v(" "),a("div",{staticClass:"field-body"},[a("b-input",{attrs:{name:"neighbor_num",size:"is-small",type:"number",min:"1"},model:{value:n.neighborNum,callback:function(e){n.neighborNum=e},expression:"neighborNum"}})],1)])]:[a("div",{staticClass:"field is-horizontal"},[a("div",{staticClass:"field-label"},[a("label",{staticClass:"label"},[n._v("Set Number of Highlighted Seed Genes\n                "),a("b-tooltip",{attrs:{label:"The random walk probability for each input seed is calculated by the random walk analysis. Ranking the probability from high to low, the top ranking seed genes will be highlighted in the network.",type:"is-light",multilined:""}},[a("b-icon",{attrs:{icon:"help-circle-outline",size:"is-small"}})],1)],1)]),n._v(" "),a("div",{staticClass:"field-body"},[a("b-input",{attrs:{name:"highlight_seed_num",size:"is-small",type:"number",min:"1"},model:{value:n.highlightSeedNum,callback:function(e){n.highlightSeedNum=e},expression:"highlightSeedNum"}})],1)])]]:n._e(),n._v(" "),a("div",{staticClass:"field is-horizontal"},[a("div",{staticClass:"field-label"},[a("label",{staticClass:"label"},[n._v("Significance Level\n            "),"NTA"!=n.enrichMethod?a("b-tooltip",{attrs:{label:"Two methods are available for identifying the enriched categories. FDR means the enriched categories will be identified based on FDR threshold and TOP means the categories will be first ranked based on the FDR and then the top most significant categories will be selected. For the GSEA analysis, WebGestalt will select the top most significant categories from each of positive and negative related categories.",type:"is-light",multilined:""}},[a("b-icon",{attrs:{icon:"help-circle-outline",size:"is-small"}})],1):a("b-tooltip",{attrs:{label:"Two methods are available for significant level method for enrichment analysis. FDR method will identify the significant GO terms based on the adjusted p value of each GO term while Top method will first rank GO terms based on the adjusted p value and select the top GO terms as the significant terms.",type:"is-light",multilined:""}},[a("b-icon",{attrs:{icon:"help-circle-outline",size:"is-small"}})],1)],1)]),n._v(" "),a("div",{staticClass:"field-body"},[a("b-field",{attrs:{grouped:""}},[a("b-radio",{attrs:{name:"sig_method","native-value":"fdr",size:"is-small"},on:{input:n.onSigChange},model:{value:n.sigMethod,callback:function(e){n.sigMethod=e},expression:"sigMethod"}},[n._v("FDR")]),n._v(" "),a("b-radio",{attrs:{name:"sig_method","native-value":"top",size:"is-small"},on:{input:n.onSigChange},model:{value:n.sigMethod,callback:function(e){n.sigMethod=e},expression:"sigMethod"}},[n._v("TOP")]),n._v(" "),"top"===n.sigMethod?[a("b-input",{ref:"sigValue",attrs:{id:"sig_value",name:"sig_value",size:"is-small",type:"number",min:"1",step:"1"},model:{value:n.sigValue,callback:function(e){n.sigValue=e},expression:"sigValue"}})]:n._e(),n._v(" "),"fdr"===n.sigMethod?[a("b-input",{ref:"sigValue",attrs:{id:"sig_value",name:"sig_value",size:"is-small",type:"number",min:"0",max:"1",step:"any"},model:{value:n.sigValue,callback:function(e){n.sigValue=e},expression:"sigValue"}})]:n._e()],2)],1)]),n._v(" "),"NTA"==n.enrichMethod&&"Network_Expansion"==n.ntaMethod?[a("div",{staticClass:"field is-horizontal"},[a("div",{staticClass:"field-label"},[a("label",{staticClass:"label"},[n._v("Highlight\n              "),a("b-tooltip",{attrs:{label:"Seeds (or neighbors) will be highlighted in the network graph (bigger nodes) and in the detailed information table (bigger font).",type:"is-light",multilined:""}},[a("b-icon",{attrs:{icon:"help-circle-outline",size:"is-small"}})],1)],1)]),n._v(" "),a("div",{staticClass:"field-body"},[a("b-field",[a("b-radio",{attrs:{name:"NTA_highlight","native-value":"Seeds",size:"is-small"},model:{value:n.ntaHighlight,callback:function(e){n.ntaHighlight=e},expression:"ntaHighlight"}},[n._v("Seeds")]),n._v(" "),a("b-radio",{attrs:{name:"NTA_highlight","native-value":"Neighbors",size:"is-small"},model:{value:n.ntaHighlight,callback:function(e){n.ntaHighlight=e},expression:"ntaHighlight"}},[n._v("Neighbors")])],1)],1)])]:n._e(),n._v(" "),"GSEA"==n.enrichMethod?[a("div",{staticClass:"field is-horizontal"},[a("div",{staticClass:"field-label"},[a("label",{staticClass:"label"},[n._v("Number of Permutations\n            "),a("b-tooltip",{attrs:{label:"The number of permutations for the GSEA method.",type:"is-light",multilined:""}},[a("b-icon",{attrs:{icon:"help-circle-outline",size:"is-small"}})],1)],1)]),n._v(" "),a("div",{staticClass:"field-body"},[a("b-input",{attrs:{name:"per_num",size:"is-small",type:"number",min:"100",max:"10000"},model:{value:n.perNum,callback:function(e){n.perNum=e},expression:"perNum"}})],1)]),n._v(" "),a("div",{staticClass:"field is-horizontal"},[a("div",{staticClass:"field-label"},[a("label",{staticClass:"label"},[n._v("p\n            "),a("b-tooltip",{attrs:{label:"Enrichment statistic. The exponential scaling factor of the phenotype score in enrichment score formula.",type:"is-light",multilined:""}},[a("b-icon",{attrs:{icon:"help-circle-outline",size:"is-small"}})],1)],1)]),n._v(" "),a("div",{staticClass:"field-body"},[a("b-select",{attrs:{name:"gsea_p",size:"is-small"},model:{value:n.gseaP,callback:function(e){n.gseaP=e},expression:"gseaP"}},[a("option",{attrs:{value:"1"}},[n._v("1")]),n._v(" "),a("option",{attrs:{value:"0"}},[n._v("0")]),n._v(" "),a("option",{attrs:{value:"1.5"}},[n._v("1.5")]),n._v(" "),a("option",{attrs:{value:"2"}},[n._v("2")])])],1)]),n._v(" "),a("div",{staticClass:"field is-horizontal"},[a("div",{staticClass:"field-label"},[a("label",{staticClass:"label"},[n._v("Collapse Method\n            "),a("b-tooltip",{attrs:{label:"The method for collaping the duplicate IDs.",type:"is-light",multilined:""}},[a("b-icon",{attrs:{icon:"help-circle-outline",size:"is-small"}})],1)],1)]),n._v(" "),a("div",{staticClass:"field-body"},[a("b-select",{attrs:{name:"collapse_method",size:"is-small"},model:{value:n.collapseMethod,callback:function(e){n.collapseMethod=e},expression:"collapseMethod"}},[a("option",{attrs:{value:"mean"}},[n._v("Mean")]),n._v(" "),a("option",{attrs:{value:"median"}},[n._v("Median")]),n._v(" "),a("option",{attrs:{value:"max"}},[n._v("Maximum")]),n._v(" "),a("option",{attrs:{value:"min"}},[n._v("Minimum")])])],1)])]:n._e(),n._v(" "),"NTA"!=n.enrichMethod?[a("div",{staticClass:"field is-horizontal"},[a("div",{staticClass:"field-label"},[a("label",{staticClass:"label"},[n._v("Number of categories expected from set cover\n            "),a("b-tooltip",{attrs:{label:"This is the number of reduced sets when set cover algorithm stops.",type:"is-light",multilined:""}},[a("b-icon",{attrs:{icon:"help-circle-outline",size:"is-small"}})],1)],1)]),n._v(" "),a("div",{staticClass:"field-body"},[a("b-input",{attrs:{name:"set_cover_num",size:"is-small",type:"number",min:"1"},model:{value:n.setCoverNum,callback:function(e){n.setCoverNum=e},expression:"setCoverNum"}})],1)]),n._v(" "),a("div",{staticClass:"field is-horizontal"},[a("div",{staticClass:"field-label"},[a("label",{staticClass:"label"},[n._v("Number of categories visualized in the report\n            "),a("b-tooltip",{attrs:{label:"This limits the number of categories shown in the report and the maximum is 100.",type:"is-light",multilined:""}},[a("b-icon",{attrs:{icon:"help-circle-outline",size:"is-small"}})],1)],1)]),n._v(" "),a("div",{staticClass:"field-body"},[a("b-input",{attrs:{name:"report_num",size:"is-small",type:"number",min:"1",max:"100"},model:{value:n.reportNum,callback:function(e){n.reportNum=e},expression:"reportNum"}})],1)]),n._v(" "),a("div",{staticClass:"field is-horizontal"},[a("div",{staticClass:"field-label"},[a("label",{staticClass:"label"},[n._v("Color in DAG\n            "),a("b-tooltip",{attrs:{label:"Use binary or continuous color scheme",type:"is-light",multilined:""}},[a("b-icon",{attrs:{icon:"help-circle-outline",size:"is-small"}})],1)],1)]),n._v(" "),a("div",{staticClass:"field-body"},[a("b-field",[a("b-radio",{attrs:{name:"color_scheme","native-value":"continuous",size:"is-small"},model:{value:n.colorScheme,callback:function(e){n.colorScheme=e},expression:"colorScheme"}},[n._v("Continuous")]),n._v(" "),a("b-radio",{attrs:{name:"color_scheme","native-value":"binary",size:"is-small"},model:{value:n.colorScheme,callback:function(e){n.colorScheme=e},expression:"colorScheme"}},[n._v("Binary")])],1)],1)])]:n._e()],2)]),n._v(" "),a("button",{staticClass:"button is-primary",attrs:{disabled:!n.isValid}},[n._v("Submit")])])},[function(){var e=this.$createElement,t=this._self._c||e;return t("div",{staticClass:"field-label is-small"},[t("label",{staticClass:"label"},[this._v("ID type of the custom database")])])},function(){var e=this.$createElement,t=this._self._c||e;return t("p",{staticClass:"title is-5 form-banner"},[t("span",[this._v("Gene List")])])},function(){var e=this.$createElement,t=this._self._c||e;return t("div",{staticClass:"field-label"},[t("label",{staticClass:"label"})])},function(){var e=this.$createElement,t=this._self._c||e;return t("p",{staticClass:"title is-5 form-banner"},[t("span",[this._v("Reference Gene List")])])}],!1,null,null,null).exports;o(45),t.default=function(e){new a.a({el:"#app",mixins:[{data:e}],data:function(){return{isCitationActive:!1}},computed:{},components:{SubmitForm:c},methods:{setOraExample:function(){var t=this;fetch("./interstingGenes.txt").then(function(e){return e.text()}).then(function(e){t.initialValues={enrichMethod:"ORA",organism:"hsapiens",databases:[{category:"geneontology",name:"Biological_Process_noRedundant"}],idType:"genesymbol",geneList:e,reference:"genome_protein-coding",minNum:5,maxNum:2e3,fdrMethod:"BH",sigMethod:"fdr",sigValue:.05,reportNum:100}})},setGseaExample:function(){var t=this;fetch("./GeneRankList.rnk").then(function(e){return e.text()}).then(function(e){t.initialValues={enrichMethod:"GSEA",organism:"hsapiens",databases:[{category:"pathway",name:"KEGG"}],idType:"genesymbol",geneList:e,minNum:20,maxNum:500,sigMethod:"top",sigValue:10,perNum:1e3,collapseMethod:"mean",reportNum:40}})},setNtaExample:function(){var t=this;fetch("./EMT.txt").then(function(e){return e.text()}).then(function(e){t.initialValues={enrichMethod:"NTA",organism:"hsapiens",databases:[{category:"network",name:"PPI_BIOGRID"}],idType:"genesymbol",ntaMethod:"Network_Retrieval_Prioritization",geneList:e,sigMethod:"top",higlightSeedNum:10,sigValue:10}})},setPhosphositeExample:function(){var t=this;fetch("./phosphosite_example.txt").then(function(e){return e.text()}).then(function(e){t.initialValues={enrichMethod:"ORA",organism:"hsapiens",databases:[{category:"network",name:"Kinase_phosphosite"}],idType:"phosphositeSeq",geneList:e,reference:"phosphosite",sigMethod:"top",minNum:3,maxNum:2e3,fdrMethod:"BH",sigValue:10,reportNum:40}})}},created:function(){}})}},5:function(e,t){e.exports=Vue}}).default;