---
title: "RNA sequencing of microdissected kidney biopsies from IgA nephropathy patients"
subtitle: "Project #6026"
author: "Vincent van Hoef"
date: today
title-block-banner: true
reference-location: margin
citation-location: margin
bibliography: ./assets/citations.bib
csl: ./assets/mla.cls
format:
    html:
        theme: cosmo
        toc: true
        toc-location: left
        css: "./assets/style.css"
        self-contained: false
        include-before-body: ./assets/custom.html
execute: 
  echo: false
  warning: false
comments: 
    hypothesis: true
---

:::{.callout-note collapse="true"}

# Click here to see the version and updates of this report

Here you can see the status of the report. This will be updated as sections are added or improved.

- Version 1: [15/7/22] First version - several sections still missing. 

:::

# Introduction

::: {.callout-note}
## Project Description
"Our aim is to unravel global molecular fingerprints of glomerular diseases by RNA profiling of kidney tissue . We sample tissue from diagnostic kidney biopsies and perform microdissection into glomerular and tubulointerstitial fractions. In this subproject, RNA isolation and library preparation has been performed (via BEA core faciltiy) on glomerular and tubulointerstitial fractions from IgAN (n=84) and healthy living kidney donors (n=11). Libraries were sequenced at SciLife lab (Illumina Novaseq 6000). Now we need help to analyse differential gene expression between patients and controls, perform enriched pathway analyses and also to correlate DEG to clinical variables."
:::

In total, there are 189 samples (One missing - which?). 
```{ojs}
samples = FileAttachment("./db/metadata.csv").csv({typed: true})
```

```{ojs}
Inputs.table(samples)
```

A contract was set up for 100 hours:

* Run best practive RNAseq preprocessing pipeline
* Pre- and post-alignment quality control
* Evaluate and if necessary control for data heterogeneity
* Differential expression of patients vs control + pathway analysis
* correlation gene expression w/ clinical parameters and pathology scores (discuss with User)
* Report with all results, tables and analysis code

# Material and Methods

Analysis code is available on Github: [link](https://github.com/vincent-van-hoef/SMS-6026-22-kidney).

## Preprocessing

Quality control was performed on raw and trimmed reads using FastQC (v0.11.8) and FastP (v0.23.2). The results for all samples were summarized using MultiQC (v1.7). Sequencing adapters and low quality bases and reads were removed using CutAdapt v2.3 (wrapped in TrimGalore v0.6.1) using the standard parameters.

## (Quasi-)Mapping

Transcript expression was quantified using Salmon (v1.6.0) and XXX collection of transcripts. Salmon was run using the --validateMappings, --seqBias and --gcBias flags. Gene-levels count estimates were collected by using the tximport R package (v1.20.0).

## Differential Expression

The R package DESeq2 (v1.32.0) was used for the differential expression analyses.

## Pathway Analysis

# Results

## Quality Control

### MultiQC

Modern high throughput sequencers can generate hundreds of millions of sequences in a single run. Before analysing this sequences to draw biological conclusions you should always perform some simple quality control checks to ensure that the raw data looks good and there are no problems or biases in your data which may affect how you can usefully use it. Two very useful quality control softwares were run on the data: FastQC and FastP. These tools are run on a per-sample basis and then summarized using MultiQC (plots below). These contain many high-quality plots; most of them are interactive and with relevant help sections to help interpretation. Most figures can be filtered and downloaded as needed. If the figures are too small to see, please right-click and “Open Frame in New Tab”

::: {.panel-tabset .column-page-inset-right}

# Raw Reads

# Trimmed Reads

:::

### PCA

```{r}
data(cars)
ojs_define(cars = cars)
```

```{ojs}
speedChart = Plot.plot({
  marks: [
  Plot.line(cars,
  {x: "speed", y: "dist"},
  {stroke: "red"})
  ]
})
```

## Differential Expression

## Pathway Analysis


# Data Responsibility

- **NBIS & Uppnex** Unfortunately, we do not have resources to keep any files associated with the support request. We suggest that you safely store the results delivered by us. In addition, we ask that you remove the files from UPPMAX/UPPNEX after analysis is completed. The main storage at UPPNEX is optimized for high-speed and parallel access, which makes it expensive and not the right place for long time archiving.
- **Sensitive data** Please note that special considerations may apply to the human-derived sensitive personal data. These should be handled according to specific laws and regulations.
- **Long-term backup** The responsibility for data archiving lies with universities and we recommend asking your local IT for support with long-term data archiving. Also the newly established Data Office at SciLifeLab may be of help to discuss other options.

# Acknowledgements

If you are presenting the results in a paper, at a workshop or conference, we kindly ask you to acknowledge us.

- **NBIS Staff** are encouraged to be co-authors when this is merited in accordance to the ethical recommendations for authorship, e.g. [ICMJE recommendations](http://www.icmje.org/recommendations/browse/roles-and-responsibilities/defining-the-role-of-authors-and-contributors.html). If applicable, please include **Vincent van Hoef, National Bioinformatics Infrastructure Sweden, Science for Life Laboratory, Uppsala University**, as co-author. In other cases, NBIS would be grateful if support by us is acknowledged in publications according to this example: ["Support by NBIS (National Bioinformatics Infrastructure Sweden) is gratefully acknowledged"](https://www.nbis.se/resources/support.html).

- **UPPMAX** If your project has used HPC resources we kindly asks you to acknowledge UPPMAX and SNIC. If applicable, please add: ["The computations were performed on resources provided by SNIC through Uppsala Multidisciplinary Center for Advanced Computational Science (UPPMAX) under Project SNIC XXXX/Y-ZZZ"](https://www.uppmax.uu.se/support/faq/general-miscellaneous-faq/acknowledging-uppmax--snic--and-uppnex/).

- **NGI** In publications based on data from NGI Sweden, the authors must acknowledge SciLifeLab, NGI and UPPMAX: ["The authors would like to acknowledge support from Science for Life Laboratory, the National Genomics Infrastructure, NGI, and Uppmax for providing assistance in massive parallel sequencing and computational infrastructure"](https://ngisweden.scilifelab.se/info/faq#how-do-i-acknowledge-ngi-in-my-publication).

# Closing Procedures

You should soon be contacted by one of our managers, [Jessica Lindvall](jessica.lindvall@nbis.se) or [Henrik Lantz](henrik.lantz@nbis.se), with a request to close down the project in our internal system and for invoicing matters. If we do not hear from you within **30 days** the project will be automatically closed and invoice sent. Again, we would like to remind you about data responsibility and acknowledgements, see Data Responsibility and Acknowledgments sections.

You are naturally more than welcome to come back to us with further data analysis request at any time via [NBIS support](http://nbis.se/support/support.html).

Thank you for using NBIS and all the best for future research.